DICOM/src/Microsoft.Health.DeID.SharedLib/Utilities/DateTimeUtility.cs

```diff
@@ -0,0 +1,45 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using Microsoft.Health.Dicom.DeID.SharedLib.Exceptions;
+
+namespace Microsoft.Health.Dicom.DeID.SharedLib
+{
+    public class DateTimeUtility
+    {
+        public static DateTimeOffset ParseDateTimeString(string inputString, string globalFormat = null)
+        {
+            if (DateTimeOffset.TryParse(inputString, out DateTimeOffset defaultResult))
+            {
+                return defaultResult;
+            }
+
+            if (DateTimeOffset.TryParseExact(inputString, globalFormat, CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTimeOffset globalResult))
+            {
+                return globalResult;
+            }
+
+            if (DateTimeOffset.TryParseExact(inputString, "yyyy", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTimeOffset yearResult))
```

Comment:
❔ [Question]

So if the DateTimeOffset cannot be parsed correctly, why do we need to try ‘yyyy’? We not ‘dddd’ or other custmerized format?